frontend/src/logic/schedule/data-row.logic.ts

```diff
@@ -59,4 +63,37 @@ export class DataRow {
   public findValues(...args: string[]): string[] {
     return args.map((arg) => this.findValue(arg));
   }
+  //#endregion
+
+  //# row check region
+  private checkShiftRowPattern(): boolean {
+    const containsNotEmptyKey = this.data[0] != null && this.data[0] != "";
+
+    let shiftCodesList = Object.keys(ShiftCode)
+        .map(k => ShiftCode[k as any])
+    let containsAnyShiftCode = false;
+    shiftCodesList.forEach(shift => {
+      if(this.data.includes(shift)){
+        containsAnyShiftCode = true;
+      }
+    })
+
+    const dataLen = this.data.length
+    const hoursCellsNumber = 3
+    if(this.data.length < hoursCellsNumber){
+      return false
+    }
+    let containsHoursInfo = true;
+    for(let i = 1; i<hoursCellsNumber + 1; i++){
+      if(typeof this.data[dataLen - i] !== 'number'){
```

Comment:
Może być tak, że ostatnia któraś z tych komórek bliżej końca będzie nullem i wtedy nie rozpoznamy tej linijki jako linijki ze zmianiami. Więc myślę, że lepiej pozbyć się ograniczenia na te 3 ostatnie komórki

Reply:
Obawiam, się tego, że np gdzieś pojawi się jakiś wiersz, gdzie będzie po prostu literka z kodem i to nam się wysypie, a ten constraint jednak daje duże zabezpieczenie, ale faktycznie może być tak jak mówisz. Wszystko zależy od tego co dostaniemy w nowych harmonogramach.

Reply:
Masz rację. Myślę, że można było by dodać warunek na to, że musi być kilka kodów w wierszu. Ale zobaczymy kiedy dostaniemy nowe harmonogramy